projects/cn0540/common/cn0540_bd.tcl

```diff
@@ -0,0 +1,117 @@
+ad_connect iic_cn0540 axi_iic_cn0540/iic
+
+#ad_connect  sys_concat_intc/In11  axi_iic_cn0540/iic2intc_irpt
+ad_cpu_interrupt "ps-8" "mb-8" axi_iic_cn0540/iic2intc_irpt
+
+# create a SPI Engine architecture for ADC
+
+create_bd_cell -type hier spi_adc
+current_bd_instance /spi_adc
+
+  create_bd_pin -dir I -type clk clk
+  create_bd_pin -dir I -type clk spi_clk
+  create_bd_pin -dir I -type rst resetn
+  create_bd_pin -dir I drdy
+  create_bd_pin -dir O irq
+  create_bd_intf_pin -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 m_spi
+  create_bd_intf_pin -mode Master -vlnv xilinx.com:interface:axis_rtl:1.0 M_AXIS_SAMPLE
+
+  # DATA_WIDTH is set to 32
+
+  ad_ip_instance spi_engine_execution execution
+  ad_ip_parameter execution CONFIG.DATA_WIDTH 32
+  ad_ip_parameter execution CONFIG.NUM_OF_CS 1
+
+  ad_ip_instance axi_spi_engine axi_regmap
+  ad_ip_parameter axi_regmap CONFIG.DATA_WIDTH 32
+  ad_ip_parameter axi_regmap CONFIG.NUM_OFFLOAD 1
+  ad_ip_parameter axi_regmap CONFIG.ASYNC_SPI_CLK 1
+
+  ad_ip_instance spi_engine_offload offload
+  ad_ip_parameter offload CONFIG.DATA_WIDTH 32
+  ad_ip_parameter offload CONFIG.ASYNC_TRIG 1
+  ad_ip_parameter offload CONFIG.ASYNC_SPI_CLK 1
+
+  ad_ip_instance spi_engine_interconnect interconnect
+  ad_ip_parameter interconnect CONFIG.DATA_WIDTH 32
+
+  ad_connect axi_regmap/spi_engine_offload_ctrl0 offload/spi_engine_offload_ctrl
+  ad_connect offload/spi_engine_ctrl interconnect/s0_ctrl
+  ad_connect axi_regmap/spi_engine_ctrl interconnect/s1_ctrl
+  ad_connect interconnect/m_ctrl execution/ctrl
+  ad_connect offload/offload_sdi M_AXIS_SAMPLE
+
+  ad_connect execution/spi m_spi
+
+  ad_connect spi_clk offload/spi_clk
+  ad_connect spi_clk offload/ctrl_clk
+  ad_connect spi_clk execution/clk
+  ad_connect clk axi_regmap/s_axi_aclk
+  ad_connect spi_clk axi_regmap/spi_clk
+  ad_connect spi_clk interconnect/clk
+
+  ad_connect axi_regmap/spi_resetn offload/spi_resetn
+  ad_connect axi_regmap/spi_resetn execution/resetn
+  ad_connect axi_regmap/spi_resetn interconnect/resetn
+
+  ad_connect drdy offload/trigger
+
+  ad_connect resetn axi_regmap/s_axi_aresetn
+  ad_connect irq axi_regmap/irq
+
+
+current_bd_instance /
+
+ad_connect adc_data_ready spi_adc/drdy
+
+# dma for the ADC
+
+ad_ip_instance axi_dmac axi_cn0540_dma
+ad_ip_parameter axi_cn0540_dma CONFIG.DMA_TYPE_SRC 1
+ad_ip_parameter axi_cn0540_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter axi_cn0540_dma CONFIG.CYCLIC 0
+ad_ip_parameter axi_cn0540_dma CONFIG.SYNC_TRANSFER_START 0
+ad_ip_parameter axi_cn0540_dma CONFIG.AXI_SLICE_SRC 0
+ad_ip_parameter axi_cn0540_dma CONFIG.AXI_SLICE_DEST 1
+ad_ip_parameter axi_cn0540_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter axi_cn0540_dma CONFIG.DMA_DATA_WIDTH_SRC 32
+ad_ip_parameter axi_cn0540_dma CONFIG.DMA_DATA_WIDTH_DEST 64
+
+ad_ip_parameter sys_ps7 CONFIG.PCW_EN_CLK2_PORT 1
+ad_ip_parameter sys_ps7 CONFIG.PCW_EN_RST2_PORT 1
+ad_ip_parameter sys_ps7 CONFIG.PCW_FPGA2_PERIPHERAL_FREQMHZ 40.0
```

Comment:
These are platform-dependent configurations, we should put them in system_bd.tcl.